### PR TITLE
Packaging changes required by #421

### PR DESCRIPTION
**Pull request by hal-ler**
Artifacts generated on all platforms:
omnisharp-win-x64-dnxcore50.zip
omnisharp-win-x64-dnx451.zip
omnisharp-win-x86-dnxcore50.zip
omnisharp-win-x86-dnx451.zip
omnisharp-linux-x64-dnxcore50.tar.gz
omnisharp-osx-x64-dnxcore50.tar.gz
omnisharp-linux-mono.tar.gz

Warning: tar.gz archives generated on Windows are missing the execute permissions for the executables they contain!

The files are located in artifacts/package as it was in the version merged by @troydai.

Appveyor publishes:
omnisharp-win-x64-dnxcore50.zip
omnisharp-win-x64-dnx451.zip
omnisharp-win-x86-dnxcore50.zip
omnisharp-win-x86-dnx451.zip

Travis-linux publishes:
omnisharp-linux-x64-dnxcore50.tar.gz
omnisharp-linux-mono.tar.gz

Travis osx publishes:
omnisharp-osx-x64-dnxcore50.tar.gz

### CODE REVIEW

**Comment by david-driscoll** [comment]
@troydai you're removing the multiple runtimes from the build since they're removing it from `dotnet restore` right?

That would make building the correct files for each platform much easier.

**Comment by hal-ler** [comment]
@david-driscoll He changed the project.json in OmniSharp to target all the run-times. This makes the default publish occur for all platforms. I'm fine with it as a good test of the build system, but I will make sure to only deploy the platform specific archives from each machine.

**Comment by hal-ler** [comment]
It should work with the last push, but I can't find any way to check the deployment on Travis.

**Comment by david-driscoll** [comment]
Without adding `tree` you can't.

**Comment by david-driscoll** [comment]
We can probably merge it and then I can fire off a release later today.

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ deploy:
   api_key:
     secure: N9hansErZKHl7G5Ed/hcBgwcvLuRjB7YAskAvSAYB+luacV6rSK7Vlm/4NyjaZCwWv5wOdBphle2S4yZLRDTdMwLrdQWwWYeZI60kE22c1amKJaf6j5ai2u/P3bt55klQ2yO2U/LacwHVoRtJlVdwSAXuDQ3zMd88VbBModQyxE=
   file_glob: true
-  file: artifacts/*.tar.gz
+  file: artifacts/package/*${TRAVIS_OS_NAME}*.tar.gz
```
Comment (david-driscoll) [inline .travis.yml]: Will this catch mono?

Comment (hal-ler) [inline .travis.yml]: I renamed mono to be linux-mono. It also makes sense as it was built on linux.

**Comment by david-driscoll** [comment]
So I realized that the branch event for appveyor will never fire... because your branch is on another fork.  Whoops, fixed that. :)

I'm going to merge and trigger and release to see what happens.

**Pull request merged**